### PR TITLE
feat: settings sections

### PR DESCRIPTION
**Pull request by ShawkyZ**
### Description

* Decouple storageProvider form Options.
* Persistence is managed by SnykOptionsManager.
* Different Dialog pages for each settings section.
* Persist settings on "Ok" btn click.

### Checklist

- [x] Tests added and all succeed
- [x] Linted
- [ ] README.md updated, if user-facing
- [ ] CHANGELOG.md updated

### Screenshots / GIFs

image
image
image
image
image
image

### CODE REVIEW

File: Snyk.VisualStudio.Extension.2022/Settings/SnykExperimentalDialogPage.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    // This method is used when the user clicks "Ok"
+    public override void SaveSettingsToStorage()
+    {
+        this.snykOptions.IgnoredIssuesEnabled = SnykExperimentalUserControl.OptionsMemento.IgnoredIssuesEnabled;
+        this.snykOptions.OpenIssuesEnabled = SnykExperimentalUserControl.OptionsMemento.OpenIssuesEnabled;
+        this.serviceProvider.SnykOptionsManager.Save(this.snykOptions);
+
+        if (LanguageClientHelper.IsLanguageServerReady())
```
Comment (ShawkyZ) [inline Snyk.VisualStudio.Extension.2022/Settings/SnykExperimentalDialogPage.cs:43]: and AutoScan is enabled

File: Snyk.VisualStudio.Extension.2022/Settings/SnykExperimentalUserControl.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        private void OnLoad(object sender, EventArgs e)
+        {
+            CheckForIgnores();
+        }
+
+        private void CheckForIgnores()
+        {
+            ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+            {
+                if (!serviceProvider.Options.ConsistentIgnoresEnabled && LanguageClientHelper.IsLanguageServerReady())
```
Comment (ShawkyZ) [inline Snyk.VisualStudio.Extension.2022/Settings/SnykExperimentalUserControl.cs:38]: also if ApiToken.IsValid()

File: Snyk.VisualStudio.Extension.2022/Settings/SnykOptionsManager.cs [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+            Logger.Information("Leave SaveAdditionalOptions method");
+        }
+
+        /// <summary>
+        /// Sace is all projects scan enabled.
+        /// </summary>
+        /// <param name="isAllProjectsEnabled">Bool param.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public async Task SaveIsAllProjectsScanEnabledAsync(bool isAllProjectsEnabled)
```
Comment (ShawkyZ) [inline Snyk.VisualStudio.Extension.2022/Settings/SnykOptionsManager.cs:181]: delete this

File: Snyk.VisualStudio.Extension.2022/Settings/SnykOptionsManager.cs [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+
+            this.SaveSettingsToFile();
+            serviceProvider.Options.InvokeSettingsChangedEvent();
+        }
+
+        /// <summary>
+        /// Get is all projects enabled.
+        /// </summary>
+        /// <returns>Bool.</returns>
+        public async Task<bool> GetIsAllProjectsEnabledAsync()
```
Comment (ShawkyZ) [inline Snyk.VisualStudio.Extension.2022/Settings/SnykOptionsManager.cs:110]: delete this

File: Snyk.VisualStudio.Extension.2022/Settings/SnykSolutionOptionsUserControl.cs [hunk 5]
```diff
@@ -49,19 +32,7 @@ private void AdditionalOptionsTextBox_TextChanged(object sender, EventArgs e)
...
-        }
-
-        private void AllProjectsCheckBox_CheckedChanged(object sender, EventArgs e)
-        {
-            if (this.serviceProvider.SolutionService.IsSolutionOpen())
-            {
-                this.userStorageSettingsService.SaveIsAllProjectsScanEnabledAsync(this.allProjectsCheckBox.Checked).FireAndForget();
-
-                this.CheckOptionConflicts();
+                this.serviceProvider.SnykOptionsManager.SaveAdditionalOptionsAsync(additionalOptions).FireAndForget();
```
Comment (ShawkyZ) [inline Snyk.VisualStudio.Extension.2022/Settings/SnykSolutionOptionsUserControl.cs:35]: change this to persist on Save

**Review by bastiandoetsch — APPROVED** [review]

**Pull request merged**